manifests/init.pp

```diff
@@ -497,7 +497,7 @@
   Variant[Undef,Integer,Pattern[/^(\d+)$/]] $rabbitmq_port = undef,
   Optional[String]   $rabbitmq_host = undef,
   Optional[String]   $rabbitmq_user = undef,
-  Optional[String]   $rabbitmq_password = undef,
+  Variant[Undef,Sensitive,String] $rabbitmq_password = undef,
```

Comment:
Actually I think we should keep Variant[Undef,Sensitive,String] to preserve backwards compatibility, have tests run without changes, and allow usage of a normal string as parameter.

My problem here is with the tests, code works:

```
Info: Applying configuration version '858c641 - Alessandro Franceschi, Thu Aug 17 16:49:10 2017 +0200 : Merge branch 'development' of github.com:example42/psick into development'
Notice: /Stage[main]/Sensu::Rabbitmq::Config/Sensu_rabbitmq_config[sensu.lab.psick.io]/password: changed [redacted] to [redacted]
Info: Class[Sensu::Rabbitmq::Config]: Scheduling refresh of Service[sensu-client]
Info: Class[Sensu::Rabbitmq::Config]: Scheduling refresh of Service[sensu-api]
Info: Class[Sensu::Rabbitmq::Config]: Scheduling refresh of Service[sensu-server]
Notice: /Stage[main]/Sensu::Client::Service/Service[sensu-client]: Triggered 'refresh' from 1 events
Notice: /Stage[main]/Sensu::Api::Service/Service[sensu-api]: Triggered 'refresh' from 1 events
Notice: /Stage[main]/Sensu::Server::Service/Service[sensu-server]: Triggered 'refresh' from 1 events
Notice: Applied catalog in 7.76 seconds
```
but I don't know how to write tests for the Sensitive type.